packages/client/hmi-client/src/components/model-template/tera-model-template-editor.vue

```diff
@@ -108,29 +119,49 @@ import type {
 	ModelTemplateJunction
 } from '@/types/model-templating';
 import * as modelTemplatingService from '@/services/model-templating';
+import SelectButton from 'primevue/selectbutton';
+import { KernelSessionManager } from '@/services/jupyter';
+import { logger } from '@/utils/logger';
 import TeraInfiniteCanvas from '../widgets/tera-infinite-canvas.vue';
 import TeraModelTemplate from './tera-model-template.vue';
 import TeraModelJunction from './tera-model-junction.vue';
 import TeraCanvasItem from '../widgets/tera-canvas-item.vue';
 
-defineProps<{
+const props = defineProps<{
 	model?: Model;
 }>();
 
+enum EditorView {
+	Template = 'Template',
+	Flat = 'Flat'
+}
+
+const kernelManager = new KernelSessionManager();
+
 let currentPortPosition: Position = { x: 0, y: 0 };
 let isMouseOverCanvas = false;
 let canvasTransform = { x: 0, y: 0, k: 1 };
 let isMouseOverPort = false;
 let junctionIdForNewEdge: string | null = null;
 
 const modelTemplateEditor = ref<ModelTemplateEditor>(
-	modelTemplatingService.emptyModelTemplateEditor()
+	modelTemplatingService.initializeModelTemplateEditor(props.model)
+);
+const flatModelTemplateEditor = ref<ModelTemplateEditor>(
+	modelTemplatingService.initializeModelTemplateEditor(props.model)
 );
```

Comment:
This initialization section is confusing. You have a watcher below that watches for `props.model` changes and react to it, yet your two editors seem to assume that props.model does not change and do no re-initialize. 

Also your editor construction accepts nulls, I think this is a bit of a paralysis by so many different options and could become error prone down the line. I think we should tighten things up so the behaviours can be inferred by other people in a more straightforward manner.

Reply:
Thanks, I added the watcher at the end so I forgot to update these refs